.github/workflows/ci.yaml

```diff
@@ -0,0 +1,155 @@
+      matrix:
+        python-version: [3.6, 3.7]
+        plex: ['unclaimed', 'claimed']
+        is-master:
+          - ${{ github.ref == 'refs/heads/master' }}
+        exclude:
+          - is-master: false
+            plex: claimed
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Cache dependencies
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-${{ matrix.python-version }}-pip-${{ hashFiles('requirements_dev.txt') }}
+        restore-keys: ${{ runner.os }}-${{ matrix.python-version }}-pip
+    - name: Install dependencies
+      run: |
+        pip install -U pip
+        pip install -r requirements_dev.txt
+        pip install -e .
+    - name: Set Plex credentials
+      if: matrix.plex == 'claimed'
+      run: |
+        echo "PLEXAPI_AUTH_MYPLEX_USERNAME=${{ secrets.PLEXAPI_AUTH_MYPLEX_USERNAME }}" > $GITHUB_ENV
+        echo "PLEXAPI_AUTH_MYPLEX_PASSWORD=${{ secrets.PLEXAPI_AUTH_MYPLEX_PASSWORD }}" >> $GITHUB_ENV
+    - name: Bootstrap Plex server (${{ matrix.plex }})
+      run: |
+        python \
+          -u tools/plex-bootstraptest.py \
+          --destination plex \
+          --advertise-ip=127.0.0.1 \
+          --bootstrap-timeout 540 \
+          --docker-tag ${{ env.PLEX_CONTAINER_TAG }} \
+          --${{ matrix.plex }}
+    - name: Main tests with ${{ matrix.plex }} server
+      env:
+        TEST_ACCOUNT_ONCE: ${{ matrix.plex == 'unclaimed' }}
+      run: |
+        pytest \
+          -rxXs \
+          --ignore=tests/test_sync.py \
+          --tb=native \
+          --verbose \
+          --cov=plexapi \
+          tests 
+    - name: Sync tests with ${{ matrix.plex }} server
+      if: matrix.plex == 'claimed'
+      env:
+        PLEXAPI_HEADER_PROVIDES: 'controller,sync-target'
+        PLEXAPI_HEADER_PLATFORM: iOS
+        PLEXAPI_HEADER_PLATFORM_VERSION: 11.4.1
+        PLEXAPI_HEADER_DEVICE: iPhone
+      run: |
+        pytest \
+          -rxXs \
+          --tb=native \
+          --verbose \
+          --cov=plexapi \
+          --cov-append \
+          tests/test_sync.py
+    - name: Unlink PMS from MyPlex account
+      if: matrix.plex == 'claimed' && always()
+      run: python -u tools/plex-teardowntest.py
+    - name: Upload coverage artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: coverage-${{ matrix.plex }}-${{ matrix.python-version }}
+        path: .coverage
+
+  coverage:
+    name: Process test coverage
+    runs-on: ubuntu-latest
+    needs: pytest
+    steps:
+    - name: Check out code from GitHub
+      uses: actions/checkout@v2
+    - name: Cache dependencies
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-${{ env.DEFAULT_PYTHON }}-pip-${{ hashFiles('requirements_dev.txt') }}
```

Comment:
Caching based on `hashFiles('requirements_dev.txt')` won't work since you're only installing `coverage`.